src/Schema/OracleSchemaManager.php

```diff
@@ -320,6 +320,109 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        if ($currentDatabase === null) {
+            throw new Exception('Invalid database name.');
+        }
+
+        $tableNames = $this->listTableNames();
+
+        $objectsByTable = [];
+
+        // Get all column definitions in one database call.
+        $sql = <<<'SQL'
+          SELECT C.*, D.COMMENTS AS COMMENTS
+            FROM ALL_TAB_COLUMNS C
+       LEFT JOIN ALL_COL_COMMENTS D ON D.OWNER = C.OWNER AND D.TABLE_NAME = C.TABLE_NAME AND
+                 D.COLUMN_NAME = C.COLUMN_NAME
+           WHERE C.OWNER =
+SQL;
+
+        $sql .= $this->_platform->quoteStringLiteral($currentDatabase);
+        $sql .= ' ORDER BY C.TABLE_NAME, C.COLUMN_ID';
+
+        $objectsByTable['columns'] = $this->getObjectRecordsByTable($sql, 'TABLE_NAME');
+
+        // Get all foreign keys definitions in one database call.
+        $sql = <<<'SQL'
+          SELECT COLS.TABLE_NAME,
+                 ALC.CONSTRAINT_NAME,
+                 ALC.DELETE_RULE,
+                 COLS.COLUMN_NAME "local_column",
+                 COLS.POSITION,
+                 R_COLS.TABLE_NAME "references_table",
+                 R_COLS.COLUMN_NAME "foreign_column"
+            FROM ALL_CONS_COLUMNS COLS
+       LEFT JOIN ALL_CONSTRAINTS ALC ON ALC.OWNER = COLS.OWNER AND ALC.CONSTRAINT_NAME = COLS.CONSTRAINT_NAME
+       LEFT JOIN ALL_CONS_COLUMNS R_COLS ON R_COLS.OWNER = ALC.R_OWNER AND
+                 R_COLS.CONSTRAINT_NAME = ALC.R_CONSTRAINT_NAME AND
+                 R_COLS.POSITION = COLS.POSITION
+                 WHERE ALC.CONSTRAINT_TYPE = 'R' AND COLS.OWNER =
+SQL;
+
+        $sql .= $this->_platform->quoteStringLiteral($currentDatabase);
+        $sql .= 'ORDER BY COLS.TABLE_NAME, COLS.CONSTRAINT_NAME, COLS.POSITION';
+
+        $objectsByTable['foreignKeys'] = $this->getObjectRecordsByTable($sql, 'TABLE_NAME');
```

Comment:
Please extract each of the sections into its own private method:
```php
private function getDatabaseColumnsByTable(string $databaseName): array;
private function getDatabaseIndexesByTable(string $databaseName): array;
private function getDatabaseForeignKeysByTable(string $databaseName): array;
```

Reply:
Done, not exactly like that, but trying to optimize across listTables and listTableDetails.